scalableminds/webknossos
# Add navigation+layout buttons for mobile

- Adds floating navigation buttons (see screenshot) for touch devices*. the buttons allow to:
  - toggle each sidebar
  - maximize a viewport (the one which was most recently touched)
  - navigate in the third dimension backwards/forwards (also depends on the most recently touched viewport). keeping the navigation buttons pressed moves the position continually (does not depend on the move-value at the moment, though)
- automatically collapses sidebars on narrow screens (the current threshold is 1080 pixel which is somewhat arbitrary, but loosely checked against https://gs.statcounter.com/screen-resolution-stats/mobile/worldwide. also it works on a Mi 8 device)

Instead of detecting touch support or mobile devices, I implemented the logic so that the floating buttons appear automatically when the user uses touch as input. Conversely, the buttons are hidden again on mouse over. This should make the UX smooth even for devices where both is possible (e.g., convertible laptops). For narrow screens (se above), the floating bar is shown automatically, but will also disappear on mouse move.

![image](https://github.com/scalableminds/webknossos/assets/2486553/b825c491-0844-4676-ad91-7cea7e42cacd)


### URL of deployed dev instance (used for testing):
- https://mobiletracing.webknossos.xyz

### Steps to test:
- open wk on your laptop and check that everything works as usual
- open wk on a tablet or mobile device and check that the floating buttons appear (either automatically or on touch; depending on screen width)
- test the floating buttons itself

### Issues:
- fixes #7158
- fixes #7159

------
(Please delete unneeded items, merge only when none are left open)
